Marlin/src/module/stepper.cpp

```diff
@@ -2018,60 +1988,53 @@ void Stepper::_set_position(const int32_t &a, const int32_t &b, const int32_t &c
   count_position[E_AXIS] = e;
 }
 
+
 /**
  * Get a stepper's position in steps.
  */
 int32_t Stepper::position(const AxisEnum axis) {
-  CRITICAL_SECTION_START;
-  const int32_t count_pos = count_position[axis];
-  CRITICAL_SECTION_END;
-  return count_pos;
```

Comment:
Sometimes we do want to get the position directly from the stepper counts. For example, when recovering the position after a hard stop. But getting the position from the planner can sometimes also make sense. But note that the stepper positions are always behind the planner positions unless the axes are not moving, so it's important to be selective in the use of either one. The stepper positions are guaranteed to correspond directly to the physical positions of the axes at the time this is called.

Reply:
Yes. But i just moved the call to the planner, and the planner is calling the stepper. Maybe we could clarify the function name.

Reply:
What i tried to achieve with the refactoring is that Marlin code has always to pass through the planner... Even if the planner just calls an Stepper method. That makes much easier to handle critical sections (if required at all).